keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -0,0 +1,421 @@
+
+### Goals
+
+- Expose device health information (served by Device Plugin or DRA) in Pod Status and events.
+
+### Non-Goals
+
+- Expose any other device information beyond the health.
+- Expose CPU assignment of the pod by CPU manager or any other resources assignment by other managers.
+
+## Proposal
+
+### PodStatus.AllocatedResourcesStatus
+
+As part of the InPlacePodVerticalScaling KEP, the two new fields were introduced in Pod Status to reflect the currently allocated resources for the Pod:
+
+```
+	// AllocatedResources represents the compute resources allocated for this container by the
+	// node. Kubelet sets this value to Container.Resources.Requests upon successful pod admission
+	// and after successfully admitting desired pod resize.
+	// +featureGate=InPlacePodVerticalScaling
+	// +optional
+	AllocatedResources ResourceList `json:"allocatedResources,omitempty" protobuf:"bytes,10,rep,name=allocatedResources,casttype=ResourceList,castkey=ResourceName"`
+
+	// Resources represents the compute resource requests and limits that have been successfully
+	// enacted on the running container after it has been started or has been successfully resized.
+	// +featureGate=InPlacePodVerticalScaling
+	// +optional
+	Resources *ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,11,opt,name=resources"`
+```
+
+One field reflects the resource requests and limits and the other actual allocated resources.
+
+This structure will contain standard resources as well as extended resources. As noted in the comment: https://github.com/kubernetes/kubernetes/pull/124227#issuecomment-2130503713, it is only logical to also include the status of those allocated resources. 
+
+The proposal is to keep this structure as-is to simplify parsing of well-known ResourceList data type by various consumers. Typical scenario would be to compare if the AllocatedResources match the desired state.
+
+The proposal is to introduce an additional field:
+
+```
+// AllocatedResourcesStatus represents the status of various resources
+// allocated for this Pod.
+AllocatedResourcesStatus ResourcesStatus 
+
+type ResourcesStatus map[ResourceName]ResourceStatus
+
+type ResourceStatus struct {
+    // map of unique Resource ID to its status with the following restrictions:
+    //  - ResourceID must uniquely identify the Resource allocated to the Pod on the Node for the lifetime of a Pod.
+    //  - ResourceID may not make sense outside the Pod. Often it will identify the resource on the Node, but not guaranteed.
+    //  - ResourceID of one Pod must not be compared with the ResourceID of other Pod.
+    // In case of Device Plugin ResourceID maps to DeviceID.
+    Resources map[ResourceID] ResourceStatus
+    
+    // allow to extend this struct in future with the overall health fields or things like Device Plugin version
+}
+
+type ResourceID string
+
+type ResourceStatus struct {
```

Comment:
This type has the same `ResourceStatus` name as the type above.

Reply:
Renamed to `ResourceHealth`